ksql-engine/src/main/java/io/confluent/ksql/services/DefaultServiceContext.java

```diff
@@ -70,26 +120,56 @@ public KafkaClientSupplier getKafkaClientSupplier() {
 
   @Override
   public SchemaRegistryClient getSchemaRegistryClient() {
-    return srClient;
+    return srClientSupplier.get();
   }
 
   @Override
   public Supplier<SchemaRegistryClient> getSchemaRegistryClientFactory() {
-    return srClientFactory;
+    return srClientSupplier;
   }
 
   @Override
   public ConnectClient getConnectClient() {
-    return connectClient;
+    return connectClientSupplier.get();
   }
 
   @Override
   public SimpleKsqlClient getKsqlClient() {
-    return ksqlClient;
+    return ksqlClientSupplier.get();
   }
 
   @Override
   public void close() {
-    adminClient.close();
+    if (((NotThreadSafeMemoizedSupplier)adminClientSupplier).isInitialized()) {
```

Comment:
What about using a boolean value `adminInitialized` that is set to true when `getAdminClient` is called? I think that would help going back to the Guava memoizer, right? Thus making this code simpler with the removal of the NotThreadSafeMemoizedSupplier.